src/TGUI/Widgets/ChildWindow.cpp

```diff
@@ -396,6 +492,38 @@ namespace tgui
                 }
             }
 
+            // Check if the minimize button was clicked
+            else if (m_minimizeButton->m_mouseDown)
+            {
+                // Check if the mouse is still on the minimize button
+                if (m_minimizeButton->mouseOnWidget(x, y))
+                {
+                    if (isSignalBound("minimized"))
+                        sendSignal("minimized", std::static_pointer_cast<ChildWindow>(shared_from_this()));
+                    else
+                    {
+                        // TODO: Texus, should an exception be thrown or just ignore this case?
```

Comment:
Just wanted to bring your attention to this and the same scenario a few lines down.